File: backend/src/main/java/com/example/stamp_app/controller/response/measuredDataGetResponse/MeasuredDataGetResponse.java
package com.example.stamp_app.controller.response.measuredDataGetResponse;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

import java.util.List;

@Getter
@AllArgsConstructor
@Schema(description = "測定結果取得レスポンス")
public class MeasuredDataGetResponse {

    @Schema(description = "SDI-12データリスト")
    private List<Sdi12DataGetResponse> sdi12Data;

    @Schema(description = "環境データリスト")
    private List<EnvironmentalDataGetResponse> environmentalData;

    @Schema(description = "電圧データ")
    private List<VoltageDataGetResponse> voltageData;

}

File: backend/src/main/java/com/example/stamp_app/controller/response/measuredDataGetResponse/MeasuredDataGetResponse.kt
package com.example.stamp_app.controller.response.measuredDataGetResponse

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "測定結果取得レスポンス")
data class MeasuredDataGetResponse(
	@Schema(description = "SDI-12データリスト")
	val sdi12Data: List<Sdi12DataGetResponse>,

	@Schema(description = "環境データリスト")
	val environmentalData: List<EnvironmentalDataGetResponse>,

	@Schema(description = "電圧データ")
	val voltageData: List<VoltageDataGetResponse>,
)
